doordeck-sdk/src/mingwMain/resources/doordeck_headless_sdk/utils/Utils.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Text;
+using System.Text.Json;
+
+namespace DoordeckHeadlessSDK.utils
+{
+    public static unsafe class Utils
+    {
+        private static readonly JsonSerializerOptions JsonSerializerOptions = new()
+        {
+            Converters = {
+                new AuthKeyResponseConverter()
+            }
+        };
+
+        public static sbyte* ToData<T>(this T input) =>
+            JsonSerializer.Serialize(input, JsonSerializerOptions).ToSByte();
+
+        public static T? FromData<T>(sbyte* input) =>
+            JsonSerializer.Deserialize<T>(ConvertSByteToString(input), JsonSerializerOptions);
+
+        public static unsafe sbyte* ToSByte(this string input)
+        {
+            byte[] bytes = Encoding.ASCII.GetBytes(input);
+            fixed (byte* po = bytes)
+            {
+                return (sbyte*)po;
+            }
+        }
+
+        public static unsafe string ConvertSByteToString(sbyte* input)
+        {
+            // Find the length of the C-style string (null-terminated)
+            int length = 0;
+            while (input[length] != 0)
+            {
+                length++;
+            }
+            // Create a byte array from the sbyte*
+            byte[] byteArray = new byte[length];
+            for (int i = 0; i < length; i++)
+            {
+                byteArray[i] = (byte)input[i];
+            }
+            // Convert the byte array to a string (assuming ASCII encoding)
+            return Encoding.ASCII.GetString(byteArray);
```

Comment:
I think you can do this:

```
using System;
using System.Runtime.InteropServices;

public static unsafe string ConvertSByteToString(sbyte* input) 
{
    return Marshal.PtrToStringAnsi((IntPtr)ptr);
}
```
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.ptrtostringansi?view=net-9.0

Reply:
Yup, I have tested it and it seems to work!